### ISSUE TITLE
OpenSeesPy compile issue with "ElasticPPcpp" in Windows 10 (tag v3.2.0)

### ISSUE THREAD

**Issue by jk7070** [OpenSees/OpenSees#577]
I have pulled down tag v3.2.0 to test if I could compile both **OpenSees** and **OpenSeesPy**. Prior to adding any User defined materials everything compiles correctly and works, I managed to work through all the linker issues (mostly related to ActiveTcl location).

My issue comes when I try to compile the typical **ElasticPPcpp** material example. I have followed the YouTube example of Prof. Scott (https://www.youtube.com/watch?v=lub-N0kzvqk) several times and I am quite confident I followed the steps correctly.

I have been successful at compiling OpenSees including the **ElasticPPcpp** user mat. But I cannot complete the compile for OpenSeesPy. Getting the following error output.

![image](https://user-images.githubusercontent.com/42515384/116791512-2fa6d100-aabb-11eb-9df2-d16eada1b58b.png)

NOTE: if you click on the image above the text becomes significantly clearer and easier to read.

Any help most appreciated.

**Comment by mhscott** [comment]
Is ElasticPPcpp.cpp included in the material project of your workspace?


From: jk7070 ***@***.***>
Sent: Saturday, May 1, 2021 11:27 AM
To: OpenSees/OpenSees ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [OpenSees/OpenSees] OpenSeesPy compile issue with "ElasticPPcpp" in Windows 10 (tag v3.2.0) (#577)


[This email originated from outside of OSU. Use caution with links and attachments.]

I have pulled down tag v3.2.0 to test if I could compile both OpenSees and OpenSeesPy. Prior to adding any User defined materials everything compiles correctly and works, I managed to work through all the linker issues (mostly related to ActiveTcl location).

My issue comes when I try to compile the typical ElasticPPcpp material example. I have followed the YouTube example of Prof. Scott (https://www.youtube.com/watch?v=lub-N0kzvqk<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3Dlub-N0kzvqk&data=04%7C01%7C%7C84133986206c4782b9ee08d90ccea543%7Cce6d05e13c5e4d6287a84c4a2713c113%7C0%7C0%7C637554903935215157%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=g1S%2Brktsg%2B2I%2FT0PhPkikvXj8FPEPRSyYYTTIrdVybw%3D&reserved=0>) several times and I am quite confident I followed the steps correctly.

I have been successful at compiling OpenSees including the ElasticPPcpp user mat. But I cannot complete the compile for OpenSeesPy. Getting the following error output.

[image]<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F42515384%2F116791512-2fa6d100-aabb-11eb-9df2-d16eada1b58b.png&data=04%7C01%7C%7C84133986206c4782b9ee08d90ccea543%7Cce6d05e13c5e4d6287a84c4a2713c113%7C0%7C0%7C637554903935225152%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=8lK%2Bsyp%2F9lPudgjQSuja9qBDhu8M2%2BjTG6e04tYIJ%2BA%3D&reserved=0>

Any help most appreciated.

-

**Comment by jk7070** [comment]
Yes, I have added both ElasticPPcpp.cpp and ElasticPPcpp.h to the uniaxial directory of the material project and I have ensured that these files have been added to material project in VisualStudio. This part at least compiles without error.

![image](https://user-images.githubusercontent.com/42515384/116808711-aa630100-ab3a-11eb-9f4d-01dd4a488d74.png)

Next, to see if the issue was in the python interpreter I have now instead tried to add the ElasticPPcpp material to the Tcl interpreter (TclModelBuilderUniaxialMaterialCommand.cpp) but I am now getting a similar error when I try to compile **OpenSees**. For the Tcl case I have added the material to the interpreter as follows:

![image](https://user-images.githubusercontent.com/42515384/116809383-0aa77200-ab3e-11eb-8fd2-9aade6b39079.png)

**Comment by jk7070** [comment]
So I have managed to get the ElasticPPcpp element to compile. The issue seems to be related to the **OPS_Export** before the 
 'void' return type in the example **ElasticPPcpp.cpp** file. 

![image](https://user-images.githubusercontent.com/42515384/116813380-d0e16600-ab53-11eb-80e1-d0c1d078a125.png)

Looking at the already included ElasticMaterial.cpp file, which doesn't have this, I decided to remove as shown below and code compiles. I was able to use the ElasticPPcpp in python.

![image](https://user-images.githubusercontent.com/42515384/116813500-55cc7f80-ab54-11eb-9fc9-2ae04db8aec0.png)
